Source/Request.swift

```diff
@@ -224,270 +217,46 @@ public class Request {
-        {
-            var disposition: Foundation.URLSession.ResponseDisposition = .allow
-
-            expectedContentLength = response.expectedContentLength
-
-            if let dataTaskDidReceiveResponse = dataTaskDidReceiveResponse {
-                disposition = dataTaskDidReceiveResponse(session, dataTask, response)
-            }
-
-            completionHandler(disposition)
-        }
-
-        func urlSession(
-            _ session: URLSession,
-            dataTask: URLSessionDataTask,
-            didBecome downloadTask: URLSessionDownloadTask)
-        {
-            dataTaskDidBecomeDownloadTask?(session, dataTask, downloadTask)
-        }
-
-        func urlSession(_ session: URLSession, dataTask: URLSessionDataTask, didReceive data: Data) {
-            if initialResponseTime == nil { initialResponseTime = CFAbsoluteTimeGetCurrent() }
-
-            if let dataTaskDidReceiveData = dataTaskDidReceiveData {
-                dataTaskDidReceiveData(session, dataTask, data)
-            } else {
-                if let dataStream = dataStream {
-                    dataStream(data: data)
-                } else {
-                    mutableData.append(data)
-                }
-
-                totalBytesReceived += data.count
-                let totalBytesExpected = dataTask.response?.expectedContentLength ?? NSURLSessionTransferSizeUnknown
-
-                progress.totalUnitCount = totalBytesExpected
-                progress.completedUnitCount = totalBytesReceived
-
-                dataProgress?(
-                    bytesReceived: Int64(data.count),
-                    totalBytesReceived: totalBytesReceived,
-                    totalBytesExpectedToReceive: totalBytesExpected
-                )
-            }
-        }
-
-        func urlSession(
-            _ session: URLSession,
-            dataTask: URLSessionDataTask,
-            willCacheResponse proposedResponse: CachedURLResponse,
-            completionHandler: ((CachedURLResponse?) -> Void))
-        {
-            var cachedResponse: CachedURLResponse? = proposedResponse
+    // MARK: Download Destination
+
+    /// Creates a download file destination closure which uses the default file manager to move the temporary file to a
+    /// file URL in the first available directory with the specified search path directory and search path domain mask.
+    ///
+    /// - parameter directory: The search path directory. `.DocumentDirectory` by default.
+    /// - parameter domain:    The search path domain mask. `.UserDomainMask` by default.
+    ///
+    /// - returns: A download file destination closure.
+    public class func suggestedDownloadDestination(
+        directory: FileManager.SearchPathDirectory = .documentDirectory,
+        domain: FileManager.SearchPathDomainMask = .userDomainMask)
+        -> DownloadFileDestination
+    {
+        return { temporaryURL, response -> URL in
```

Comment:
I'm surprised the compiler doesn't require enclosing capture parameters in `()`. Perhaps they should be?